ExtractUserDataActivity/handler.ts

```diff
@@ -432,6 +464,71 @@ export const createExtractUserDataActivityHandler = (
         }
       );
 
+  /**
+   * Creates a bundle with all user data and save it to a blob on a remote storage
+   * @param data all extracted user data
+   * @param password a password for bundle encryption
+   *
+   * @returns either a failure or an object with the name of the blob and the password
+   */
+  const saveDataToBlob = (
+    data: AllUserData,
+    password: StrongPassword
+  ): TaskEither<ActivityResultArchiveGenerationFailure, ArchiveInfo> =>
+    taskify(
+      (
+        cb: (
+          e: ActivityResultArchiveGenerationFailure | null,
+          r?: ArchiveInfo
+        ) => void
+      ) => {
+        const blobName = `${
+          data.profile.fiscalCode
+        }-${Date.now()}.zip` as NonEmptyString;
+        const fileName = `${data.profile.fiscalCode}.json` as NonEmptyString;
+
+        const readableZipStream = createCompressedStream(
+          {
+            [fileName]: data
+          },
+          password
+        );
+
+        const writableBlobStream = blobService.createWriteStreamToBlockBlob(
+          userDataContainerName,
+          blobName,
+          (err, _) => {
+            if (err) {
+              cb(
+                ActivityResultArchiveGenerationFailure.encode({
+                  kind: "ARCHIVE_GENERATION_FAILURE",
+                  reason: err.message
+                })
+              );
+            } else {
+              cb(
+                null,
+                ArchiveInfo.encode({
+                  blobName,
+                  password
+                })
+              );
+            }
+          }
+        );
+        readableZipStream.pipe(writableBlobStream);
```

Comment:
pipe should be called before any operation on the stream (append / finalize)

Reply:
It is _implicity_ true already as `append` and `finalize` from the `archiver` module operate on another eventloop iteration. I agree we should write explicit code 
I see three options:
1. we use the `archiver-append` function outside of the `zip` module, so we can call it after the zip stream has been piped. This leads to a leak of the archiver api to the activity handler, which may not know about it
2. we keep the `archiver-append` call inside the `zip` module, as it is now, but we enforce a `process.nextTick`
3. (don't know if it works) we keep the `archiver-append` call inside the `zip` module, but we wait for the `pipe` event of the readable stream (so we're sure we won't load files before it has been piped to somewhere)

Keep in mind that the main reason of having a `zip` module is that module initialization must happen _once per application_, otherwise it would throw.

Reply:
I'd simply put all the implementation here and remove the zip module. importing modules must not have side effects, even more if this may cause an exception. if you need to initialize something, implement some init() method and make it safe.